File: java-resend-examples/src/main/java/com/resend/examples/WithAttachments.java
package com.resend.examples;

import com.resend.Resend;
import com.resend.services.emails.model.Attachment;
import com.resend.services.emails.model.CreateEmailOptions;
import com.resend.services.emails.model.CreateEmailResponse;
import io.github.cdimascio.dotenv.Dotenv;

import java.time.Instant;
import java.util.Base64;
import java.util.List;

public class WithAttachments {
    public static void main(String[] args) {
        Dotenv dotenv = Dotenv.configure().ignoreIfMissing().load();

        String apiKey = dotenv.get("RESEND_API_KEY");
        if (apiKey == null || apiKey.isEmpty()) {
            System.err.println("RESEND_API_KEY environment variable is required");
            System.exit(1);
        }

        Resend resend = new Resend(apiKey);

        String from = dotenv.get("EMAIL_FROM", "Acme <onboarding@resend.dev>");

        // Create sample file content
        String fileContent = "Sample Attachment\n==================\n\nThis file was attached to your email.\nSent at: " + Instant.now() + "\n";
        String encoded = Base64.getEncoder().encodeToString(fileContent.getBytes());

        // Maximum total attachment size: 40MB
        Attachment attachment = Attachment.builder()
                .fileName("sample.txt")
                .content(encoded)
                .build();

        CreateEmailOptions params = CreateEmailOptions.builder()
                .from(from)
                .to("delivered@resend.dev")
                .subject("Email with Attachment - Java Example")
                .html("<h1>Your attachment is ready</h1><p>Please find the file attached to this email.</p>")
                .attachments(List.of(attachment))
                .build();

        try {
            CreateEmailResponse response = resend.emails().send(params);
            System.out.println("Email with attachment sent successfully!");
            System.out.println("Email ID: " + response.getId());
        } catch (Exception e) {
            System.err.println("Error sending email: " + e.getMessage());
            System.exit(1);
        }
    }
}

File: bun-resend-examples/typescript/examples/with-attachments.ts
import { Resend } from "resend";

const resend = new Resend(process.env.RESEND_API_KEY);

const from = process.env.EMAIL_FROM || "Acme <onboarding@resend.dev>";

// Create sample file content
const fileContent = `Sample Attachment\n==================\n\nThis file was attached to your email.\nSent at: ${new Date().toISOString()}\n`;
const encoded = Buffer.from(fileContent).toString("base64");

// Maximum total attachment size: 40MB
const { data, error } = await resend.emails.send({
  from,
  to: ["delivered@resend.dev"],
  subject: "Email with Attachment",
  html: "<h1>Your attachment is ready</h1><p>Please find the file attached to this email.</p>",
  attachments: [
    {
      filename: "sample.txt",
      content: encoded,
    },
  ],
});

if (error) {
  console.error("Error sending email:", error);
  process.exit(1);
}

console.log("Email with attachment sent successfully!");
console.log("Email ID:", data?.id);
